src/server.prod.js

```diff
@@ -45,21 +48,37 @@ class ProdServer {
     const nockPath = this.config.nockPath;
     const accessLogs = this.config.accessLogs || 'clf';
     const renderFn = this.getRenderFunction();
+    const compressionMiddleware = this.config.compressHTML ? compression : () => (...[,,next]) => next();
     if (!this.config.isCDN) {
       this.setupStaticFiles(app);
     };
     app.use('/healthcheck', healthcheck());
-    app.use(...getMiddleWares({ renderFn, before, after, nock, nockPath, accessLogs }));
+    app.use(...getMiddleWares({
+      renderFn,
+      before,
+      after: [
+        ...after,
+        compressionMiddleware({
+          threshold: process.env.NODE_ENV === 'test' ? 0 : 1024,
+        })
+      ],
+      nock,
+      nockPath,
+      accessLogs
+    }));
     return app;
   }
 
   setupStaticFiles(app) {
+    const compressionMiddleware = this.config.compressAssets ? compression : () => (...[,,next]) => next();
+
     const serverPublicPath = this.config.server.output.publicPath;
     const clientPublicPath = this.config.client.output.publicPath;
     const serverOutputPath = this.config.server.output.path;
     const clientOutputPath = this.config.client.output.path;
-    app.use(serverPublicPath, express.static(serverOutputPath));
-    app.use(clientPublicPath, express.static(clientOutputPath));
+
+    app.use(serverPublicPath, compressionMiddleware(), serveStatic(serverOutputPath));
```

Comment:
Can I ask why `express.static` has been replaced by `serve-static`? Does the built-in module not support compression?

Reply:
In fact it is the same: https://github.com/expressjs/express/blob/b8e50568af9c73ef1ade434e92c60d389868361d/lib/express.js#L81

Changed.